### PR TITLE
EOS-15325 (sms20): Updating glusterfs service relation with salt-master

### PR DESCRIPTION
**Pull request by ypise**
Signed-off-by: sradhanandpati sradhanand.pati@seagate.com
Signed-off-by: Yashodhan Pise yashodhan.pise@seagate.com

### CODE REVIEW

File: cli/src/replace_node/deploy-replacement [hunk 1]
```diff
@@ -162,6 +162,7 @@ function system_states {
         # "system.storage.multipath.install"
         # "system.storage.multipath.config"
         "system.network"
+        "system.network.mgmt.service_port"
```
Comment (andkononykhin2) [inline cli/src/replace_node/deploy-replacement:165]: @ypise  looks like not related thing, do we have a confidence here?

**Review by andkononykhin2 — APPROVED** [review]

File: cli/src/replace_node/deploy-replacement (hunk 1 above)
Comment (ypise) [reply cli/src/replace_node/deploy-replacement:165]: Yes, it a required feature.
We are not performing boxing-unboxing after a node replacement. Hence, we need this.

It's also present: https://github.com/Seagate/cortx-prvsnr/blob/f1151526312d2689b7f625b8b4b507ebebd13eac/cli/src/factory_ops/boxing/prov_tasks#L115
```
function configure_service_port {
    salt "srvnode-1" state.apply components.system.network.mgmt.service_port ${salt_opts} || (
        echo "ERROR: Failed to configure service port on srvnode-1." | tee -a ${LOG_FILE}
        exit 25
    )
}
```

File: srv/components/provisioner/config.sls [hunk 2]
```diff
@@ -28,8 +28,13 @@ provisioner_rsyslog_conf_updated:
     - watch_in:
       - service: Start rsyslog
 
-# Always start glusterfshsaredstorage before salt-minion
-Update glusterfshsaredstorage.service:
+# Always start glusterfshsaredstorage before salt-master
```
Comment (nitin-seagate) [inline srv/components/provisioner/config.sls:31]: Sequence-wise should we write GlusterFS info before this Salt-Master section

**Review by nitin-seagate — APPROVED** [review]
LGTM

**Review by ujjwalpl — APPROVED** [review]

File: srv/components/provisioner/config.sls (hunk 2 above)
Comment (ypise) [reply srv/components/provisioner/config.sls:31]: @nitin-seagate here were are just copying service files from template location to target location.
No service is being restarted.

However, I've sequenced the states appropriately.

**Review by 83bhp — APPROVED** [review]
